From 2ce1319dbc7691e6b3c3689ba6b05333ffe9bf13 Mon Sep 17 00:00:00 2001
From: Thu Nguyen <thu@os.amperecomputing.com>
Date: Fri, 1 Aug 2025 22:00:52 +0000
Subject: [PATCH] user_channel: Move the D-Bus defined to common file

As [1], the "D-Bus & Object Manager related stuff" will be moved to the
common file `user_channel/user_mgmt.hpp`.

[1] https://github.com/openbmc/phosphor-host-ipmid/blob/e58eb308e561bcc0ddb6ad9cf479e35b214ab46d/user_channel/user_mgmt.cpp#L44

Upstream-Status: Pending

Change-Id: Ib98dbf7ef568a83f813023a5e9d484343af7c778
Signed-off-by: Thu Nguyen <thu@os.amperecomputing.com>
---
 user_channel/user_mgmt.cpp | 60 --------------------------------------
 user_channel/user_mgmt.hpp | 58 ++++++++++++++++++++++++++++++++++++
 2 files changed, 58 insertions(+), 60 deletions(-)

diff --git a/user_channel/user_mgmt.cpp b/user_channel/user_mgmt.cpp
index 8c5a1bf..7c30049 100644
--- a/user_channel/user_mgmt.cpp
+++ b/user_channel/user_mgmt.cpp
@@ -41,66 +41,6 @@
 namespace ipmi
 {
 
-// TODO: Move D-Bus & Object Manager related stuff, to common files
-// D-Bus property related
-static constexpr const char* dBusPropertiesInterface =
-    "org.freedesktop.DBus.Properties";
-static constexpr const char* getAllPropertiesMethod = "GetAll";
-static constexpr const char* propertiesChangedSignal = "PropertiesChanged";
-static constexpr const char* setPropertiesMethod = "Set";
-
-// Object Manager related
-static constexpr const char* dBusObjManager =
-    "org.freedesktop.DBus.ObjectManager";
-static constexpr const char* getManagedObjectsMethod = "GetManagedObjects";
-// Object Manager signals
-static constexpr const char* intfAddedSignal = "InterfacesAdded";
-static constexpr const char* intfRemovedSignal = "InterfacesRemoved";
-
-static constexpr const char* ipmiUserMutex = "ipmi_usr_mutex";
-static constexpr const char* ipmiMutexCleanupLockFile =
-    "/run/ipmi/ipmi_usr_mutex_cleanup";
-static constexpr const char* ipmiUserSignalLockFile =
-    "/run/ipmi/ipmi_usr_signal_mutex";
-static constexpr const char* ipmiUserDataFile = "/var/lib/ipmi/ipmi_user.json";
-static constexpr const char* ipmiGrpName = "ipmi";
-static constexpr size_t privNoAccess = 0xF;
-static constexpr size_t privMask = 0xF;
-
-// User manager related
-static constexpr const char* userMgrService =
-    "xyz.openbmc_project.User.Manager";
-static constexpr const char* userMgrObjBasePath = "/xyz/openbmc_project/user";
-static constexpr const char* userObjBasePath = "/xyz/openbmc_project/user";
-static constexpr const char* userMgrInterface =
-    "xyz.openbmc_project.User.Manager";
-static constexpr const char* usersInterface =
-    "xyz.openbmc_project.User.Attributes";
-static constexpr const char* deleteUserInterface =
-    "xyz.openbmc_project.Object.Delete";
-
-static constexpr const char* createUserMethod = "CreateUser";
-static constexpr const char* deleteUserMethod = "Delete";
-static constexpr const char* renameUserMethod = "RenameUser";
-// User manager signal memebers
-static constexpr const char* userRenamedSignal = "UserRenamed";
-// Mgr interface properties
-static constexpr const char* allPrivProperty = "AllPrivileges";
-static constexpr const char* allGrpProperty = "AllGroups";
-// User interface properties
-static constexpr const char* userPrivProperty = "UserPrivilege";
-static constexpr const char* userGrpProperty = "UserGroups";
-static constexpr const char* userEnabledProperty = "UserEnabled";
-
-static std::array<std::string, (PRIVILEGE_OEM + 1)> ipmiPrivIndex = {
-    "priv-reserved", // PRIVILEGE_RESERVED - 0
-    "priv-callback", // PRIVILEGE_CALLBACK - 1
-    "priv-user",     // PRIVILEGE_USER - 2
-    "priv-operator", // PRIVILEGE_OPERATOR - 3
-    "priv-admin",    // PRIVILEGE_ADMIN - 4
-    "priv-custom"    // PRIVILEGE_OEM - 5
-};
-
 using namespace phosphor::logging;
 using Json = nlohmann::json;
 
diff --git a/user_channel/user_mgmt.hpp b/user_channel/user_mgmt.hpp
index 74166c2..3f4b265 100644
--- a/user_channel/user_mgmt.hpp
+++ b/user_channel/user_mgmt.hpp
@@ -28,6 +28,64 @@
 namespace ipmi
 {
 
+static constexpr const char* dBusPropertiesInterface =
+    "org.freedesktop.DBus.Properties";
+static constexpr const char* getAllPropertiesMethod = "GetAll";
+static constexpr const char* propertiesChangedSignal = "PropertiesChanged";
+static constexpr const char* setPropertiesMethod = "Set";
+
+// Object Manager related
+static constexpr const char* dBusObjManager =
+    "org.freedesktop.DBus.ObjectManager";
+static constexpr const char* getManagedObjectsMethod = "GetManagedObjects";
+// Object Manager signals
+static constexpr const char* intfAddedSignal = "InterfacesAdded";
+static constexpr const char* intfRemovedSignal = "InterfacesRemoved";
+
+static constexpr const char* ipmiUserMutex = "ipmi_usr_mutex";
+static constexpr const char* ipmiMutexCleanupLockFile =
+    "/run/ipmi/ipmi_usr_mutex_cleanup";
+static constexpr const char* ipmiUserSignalLockFile =
+    "/run/ipmi/ipmi_usr_signal_mutex";
+static constexpr const char* ipmiUserDataFile = "/var/lib/ipmi/ipmi_user.json";
+static constexpr const char* ipmiGrpName = "ipmi";
+static constexpr size_t privNoAccess = 0xF;
+static constexpr size_t privMask = 0xF;
+
+// User manager related
+static constexpr const char* userMgrService =
+    "xyz.openbmc_project.User.Manager";
+static constexpr const char* userMgrObjBasePath = "/xyz/openbmc_project/user";
+static constexpr const char* userObjBasePath = "/xyz/openbmc_project/user";
+static constexpr const char* userMgrInterface =
+    "xyz.openbmc_project.User.Manager";
+static constexpr const char* usersInterface =
+    "xyz.openbmc_project.User.Attributes";
+static constexpr const char* deleteUserInterface =
+    "xyz.openbmc_project.Object.Delete";
+
+static constexpr const char* createUserMethod = "CreateUser";
+static constexpr const char* deleteUserMethod = "Delete";
+static constexpr const char* renameUserMethod = "RenameUser";
+// User manager signal memebers
+static constexpr const char* userRenamedSignal = "UserRenamed";
+// Mgr interface properties
+static constexpr const char* allPrivProperty = "AllPrivileges";
+static constexpr const char* allGrpProperty = "AllGroups";
+// User interface properties
+static constexpr const char* userPrivProperty = "UserPrivilege";
+static constexpr const char* userGrpProperty = "UserGroups";
+static constexpr const char* userEnabledProperty = "UserEnabled";
+
+static std::array<std::string, (PRIVILEGE_OEM + 1)> ipmiPrivIndex = {
+    "priv-reserved", // PRIVILEGE_RESERVED - 0
+    "priv-callback", // PRIVILEGE_CALLBACK - 1
+    "priv-user",     // PRIVILEGE_USER - 2
+    "priv-operator", // PRIVILEGE_OPERATOR - 3
+    "priv-admin",    // PRIVILEGE_ADMIN - 4
+    "priv-custom"    // PRIVILEGE_OEM - 5
+};
+
 using DbusUserPropVariant =
     std::variant<std::vector<std::string>, std::string, bool>;
 
-- 
2.34.1

